packages/aws-cdk/lib/api/evaluate-cloudformation-template.ts

```diff
@@ -45,6 +77,21 @@ export class EvaluateCloudFormationTemplate {
     this.urlSuffix = props.urlSuffix;
   }
 
+  public async establishResourcePhysicalName(logicalId: string, physicalNameInCfnTemplate: any): Promise<string | undefined> {
+    if (physicalNameInCfnTemplate != null) {
+      try {
+        return await this.evaluateCfnExpression(physicalNameInCfnTemplate);
+      } catch (e) {
+        // If we can't evaluate the resource's name CloudFormation expression,
```

Comment:
I think the intent here was to avoid a lookup if we could ( cc @skinny85 )

Reply:
Yes, this is used in case we can get the physical name from the template, and in this way save a service call (not super relevant here, but very relevant for hotswapping, and this class is used in both places).